File: prm393_g3_frontend/lib/features/brands/domain/entities/brand.dart
import 'package:equatable/equatable.dart';

class Brand extends Equatable {
  const Brand({
    required this.id,
    required this.name,
    this.status,
  });

  final String id;
  final String name;
  final String? status; // 'ACTIVE' or 'INACTIVE'

  @override
  List<Object?> get props => [id, name, status];
}

File: prm393_g3_backend/src/models/brand.js
import mongoose from 'mongoose';

const brandSchema = new mongoose.Schema(
  {
    name: { type: String, required: true },
    status: { type: String, enum: ['ACTIVE', 'INACTIVE'], default: 'ACTIVE' },
  },
  { timestamps: true },
);

// Explicitly map to the 'brands' collection
const Brand = mongoose.model('brand', brandSchema, 'brands');

export default Brand;